src/app/technical-record-search/test-result.service.ts

```diff
@@ -15,21 +19,87 @@ export class TestResultService {
   private readonly routes;
 
-  constructor(private httpClient: HttpClient,
+  constructor(
+    private httpClient: HttpClient,
     private _store: Store<IAppState>,
-    private _appConfig: AppConfig) {
+    private _appConfig: AppConfig
+  ) {
     this.routes = {
-      testResults: (searchIdentifier: string) => `${this._apiServer.APITestResultServerUri}/test-results/${searchIdentifier}`,
+      testResults: (searchIdentifier: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${searchIdentifier}`,
+      preparers: () => `${this._apiServer.APIPreparersServerUri}/preparers`,
+      testStation: () => `${this._apiServer.APITestStationsServerUri}/test-stations`,
+      updateTestResults: (systemNumber: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${systemNumber}`,
+      testResultById: (systemNumber: string, testResultId: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${systemNumber}?testResultId=${testResultId}&version=all`
     };
   }
 
-  getTestResults(searchIdentifier: string): Observable<any> {
+  getTestResults(searchIdentifier: string): Observable<TestResultModel> {
     this._store.dispatch(new LoadingTrue());
     const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
-    return this.httpClient.get<any[]>(this.routes.testResults(searchIdentifier), { headers }).pipe(
-      delayedRetry(),
-      shareReplay(),
-      finalize(() => this._store.dispatch(new LoadingFalse()))
-    );
+    return this.httpClient
+      .get<TestResultModel[]>(this.routes.testResults(searchIdentifier), { headers })
+      .pipe(
+        delayedRetry(),
+        shareReplay(),
+        finalize(() => this._store.dispatch(new LoadingFalse()))
+      );
+  }
+
+  getPreparers(): Observable<Preparer[]> {
+    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
+    this._store.dispatch(new LoadingTrue());
+    return this.httpClient
+      .get<Preparer[]>(this.routes.preparers(), { headers })
+      .pipe(
+        delayedRetry(),
+        shareReplay(),
+        finalize(() => this._store.dispatch(new LoadingFalse()))
+      );
+  }
+
+  getTestStations(): Observable<TestStation[]> {
+    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
+    this._store.dispatch(new LoadingTrue());
+    return this.httpClient
+      .get<TestStation[]>(this.routes.testStation(), { headers })
+      .pipe(
+        delayedRetry(),
+        shareReplay(),
+        finalize(() => this._store.dispatch(new LoadingFalse()))
+      );
+  }
+
+  updateTestResults(
+    systemNumber: string,
+    testResult: VehicleTestResultUpdate
+  ): Observable<TestResultModel> {
+    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
+    this._store.dispatch(new LoadingTrue());
+    return this.httpClient
```

Comment:
same

Reply:
As agreed with the team, the code in place was prior to ADR and a technical task has been added to trim the effects where possible
Reference: https://jira.dvsacloud.uk/browse/CVSB-15100